core/src/HIP/Kokkos_HIP_Instance.hpp

```diff
@@ -128,6 +131,161 @@ class HIPInternal {
 
   HIPInternal() = default;
 
+  // Using HIP API function/objects will be w.r.t. device 0 unless
+  // hipSetDevice(device_id) is called with the correct device_id.
+  // The correct device_id is stored in the variable
+  // HIPInternal::m_hipDev set in HIP::impl_initialize(). In the case
+  // where multiple HIP instances are used, or threads are launched
+  // using non-default HIP execution space after initialization, all HIP
+  // API calls and uses of hipStream_t variables must follow a call to
+  // hipSetDevice(device_id) when an execution space or HIPInternal
+  // object is provided to ensure all computation is done on the correct device.
+
+  // Set the device in to the device stored by this instance for HIP API calls.
+  void set_hip_device() const {
+    verify_is_initialized("set_hip_device");
+    KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(m_hipDev));
+  }
+
+  // Return this instances stream after setting the correct device
+  hipStream_t get_stream() const {
+    set_hip_device();
+    return m_stream;
+  }
+
+  // HIP API wrappers where we set the correct device id before calling the HIP
+  // API functions. For HIP API functions that take a stream, an optional input
+  // stream is available. If no stream is given, the stream for this instance is
+  // used.
+
+  // Helper function for selecting correct input stream
+  hipStream_t get_input_stream(hipStream_t s) const {
```

Comment:
I don't understand why this function and `get_input_stream` exist. Neither of this function is necessary. They just obfuscate the code.

Reply:
This is left over from when we were calling HIP API wrapper function from the singleton, where we might need to pass in a stream. I removed it and we now assume wrapper want to use the internal stream.